### ISSUE TITLE
Missing create_schema calls in schema.rb

### ISSUE THREAD

**Issue by ryanb** [rails-on-services/apartment#303]
## Steps to reproduce

Rails schema dump doesn't include `create_schema` calls in upcoming release due to #276.

## Expected behavior

It would be nice if this were configurable. We have a few schemas that exist outside tenants that need to be included in `schema.rb` dump.

The only work-around appears to be to monkey-patch this monkey-patch to reverse the monkey-patch. 😉

## Actual behavior

There's no way to configure it.

Let me know if you would like a PR to add a config option for this.

Thanks for your work on this gem! Looking forward to the next release.

**Comment by mnovelo** [comment]
I think that's reasonable. Happy to look over a PR, but wait until after the next release to create it since I'm refactoring a lot of the specs right now. I should have the next release done soon

**Comment by ryanb** [comment]
Thanks. If others need this same thing in the meantime, putting this in an initializer works to bring back the `create_schema` lines.

```rb
class ActiveRecord::ConnectionAdapters::PostgreSQL::SchemaDumper
  def schemas(stream)
    _original_schemas(stream)
  end
end
```

**Comment by mnovelo** [comment]
I plan to include this in the next version of the apartment gem

**Comment by mnovelo** [comment]
This was not included in 3.2.0, but my hope is to include this in the next major version which I'm actively working on right now

**Comment by mnovelo** [comment]
also, I'd love your input in https://github.com/rails-on-services/apartment/discussions/312

**Comment by ryanb** [comment]
Thanks @mnovelo, I appreciate your work on this!

**Comment by ryanb** [comment]
@mnovelo is this accurate? Is it not planned?

**Comment by mnovelo** [comment]
No, I just have been slow to update on my progress. This is still planned

**Comment by mnovelo** [comment]
The suppression you were working around — the #276 patch that stripped `create_schema` from the dump — was v3 code. v4 doesn't carry it: the gem no longer overrides Rails' schema dumping except for one narrow Rails 8.1 fix that strips the `public.` table prefix (#358). So stock Rails dumping applies unmodified, including its native `create_schema` emission for non-tenant schemas — the lines you wanted are back without the initializer hack.

On top of that, `pg.include_schemas_in_dump = %w[shared analytics]` (Rails 8.1+) keeps those schemas' tables schema-qualified in the dump rather than stripping the prefix.

@ryanb — could you confirm your out-of-tenant schemas dump as expected on a v4 alpha? If a specific one still goes missing, I'll treat that as a live bug. Closing as addressed in v4.
